agent/check_test.go

```diff
@@ -512,260 +509,288 @@ func TestCheckTCPPassing(t *testing.T) {
+					fmt.Fprint(w, "OK")
+				},
+				"GET /exec/456/json": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(404)
+				},
+			},
+			out:   regexp.MustCompile("^inspect exec failed for container 123: invalid exec id 456$"),
+			state: api.HealthCritical,
+		},
+		{
+			desc: "inspect exec: bad status code",
+			handlers: map[string]http.HandlerFunc{
+				"POST /containers/123/exec": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(201)
+					w.Header().Set("Content-Type", "application/json")
+					fmt.Fprint(w, `{"Id":"456"}`)
+				},
+				"POST /exec/456/start": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(200)
+					fmt.Fprint(w, "OK")
+				},
+				"GET /exec/456/json": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(999)
+					fmt.Fprint(w, "some output")
+				},
+			},
+			out:   regexp.MustCompile("^inspect exec failed for container 123 with status 999: some output$"),
+			state: api.HealthCritical,
+		},
+		{
+			desc: "inspect exec: bad json",
+			handlers: map[string]http.HandlerFunc{
+				"POST /containers/123/exec": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(201)
+					w.Header().Set("Content-Type", "application/json")
+					fmt.Fprint(w, `{"Id":"456"}`)
+				},
+				"POST /exec/456/start": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(200)
+					fmt.Fprint(w, "OK")
+				},
+				"GET /exec/456/json": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(200)
+					w.Header().Set("Content-Type", "application/json")
+					fmt.Fprint(w, `this is not json`)
+				},
+			},
+			out:   regexp.MustCompile("^inspect exec response for container 123 cannot be parsed: .*$"),
+			state: api.HealthCritical,
+		},
+		{
+			desc: "inspect exec: exit code 0: passing",
+			handlers: map[string]http.HandlerFunc{
+				"POST /containers/123/exec": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(201)
+					w.Header().Set("Content-Type", "application/json")
+					fmt.Fprint(w, `{"Id":"456"}`)
+				},
+				"POST /exec/456/start": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(200)
+					fmt.Fprint(w, "OK")
+				},
+				"GET /exec/456/json": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(200)
+					w.Header().Set("Content-Type", "application/json")
+					fmt.Fprint(w, `{"ExitCode":0}`)
+				},
+			},
+			out:   regexp.MustCompile("^OK$"),
+			state: api.HealthPassing,
+		},
+		{
+			desc: "inspect exec: exit code 0: passing: truncated",
+			handlers: map[string]http.HandlerFunc{
+				"POST /containers/123/exec": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(201)
+					w.Header().Set("Content-Type", "application/json")
+					fmt.Fprint(w, `{"Id":"456"}`)
+				},
+				"POST /exec/456/start": func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(200)
+					fmt.Fprint(w, "01234567890123456789OK") // more than 10 bytes
```

Comment:
You probably meant "20 bytes" here, since that's the output buffer size for this test.